File: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/examples-js/ListSiteDeploymentStatusSlot.js
const { WebSiteManagementClient } = require("@azure/arm-appservice");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to List deployment statuses for an app (or deployment slot, if specified).
 *
 * @summary List deployment statuses for an app (or deployment slot, if specified).
 * x-ms-original-file: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/examples/ListSiteDeploymentStatusSlot.json
 */
async function listDeploymentStatusSlot() {
  const subscriptionId = "34adfa4f-cedf-4dc0-ba29-b6d1a69ab345";
  const resourceGroupName = "rg";
  const name = "testSite";
  const slot = "stage";
  const credential = new DefaultAzureCredential();
  const client = new WebSiteManagementClient(credential, subscriptionId);
  const resArray = new Array();
  for await (let item of client.webApps.listSlotSiteDeploymentStatusesSlot(
    resourceGroupName,
    name,
    slot
  )) {
    resArray.push(item);
  }
  console.log(resArray);
}

listDeploymentStatusSlot().catch(console.error);

File: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/examples-java/ListSiteDeploymentStatusSlot.java
import com.azure.core.util.Context;

/** Samples for WebApps ListSlotSiteDeploymentStatusesSlot. */
public final class Main {
    /*
     * x-ms-original-file: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/examples/ListSiteDeploymentStatusSlot.json
     */
    /**
     * Sample code: List Deployment Status Slot.
     *
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void listDeploymentStatusSlot(com.azure.resourcemanager.AzureResourceManager azure) {
        azure
            .webApps()
            .manager()
            .serviceClient()
            .getWebApps()
            .listSlotSiteDeploymentStatusesSlot("rg", "testSite", "stage", Context.NONE);
    }
}
